### PR TITLE
[Fleet Integration] adding support for fleet gitRepo creation. Mimics clusters.go

### PR DESCRIPTION
**Pull request by slickwarren**
supports: https://github.com/rancher/qa-tasks/issues/1514

### CODE REVIEW

File: extensions/fleet/fleet.go
Comment (anupama2501) [inline extensions/fleet/fleet.go:18]: As CreateFleetGitRepo function involves creating a Fleet GitRepo and includes a retry and cleanup logic, IMO this should be considered an action rather than an extension. 

Could you let me know why we want to have this as an extension? Understanding the reasoning would help clarify whether thats the best approach.

Comment (slickwarren) [reply extensions/fleet/fleet.go:18]: messaged offline, lmk what you think but tldr is I think this still fits the definition of extension

Comment (anupama2501) [reply extensions/fleet/fleet.go:18]: discussed offline and since the gitrepo and client are the only parameters required and no additional config is required now or in future, we can have this in extensions. thanks! resolving the comment.

File: extensions/fleet/fleet.go
Comment (anupama2501) [inline extensions/fleet/fleet.go:16]: is this applicable only for rke2 or it is applicable for k3s as well?

File: extensions/fleet/fleet.go
Comment (anupama2501) [inline extensions/fleet/fleet.go:17]: nit, should be `ensure the gitrepo object is removed` instead of `cluster is removed`.

**Review by anupama2501 — COMMENTED** [review]
few nits looks good otherwise

**Review by rancher-max — DISMISSED** [review]

File: extensions/fleet/fleet.go
Comment (anupama2501) [inline extensions/fleet/fleet.go:17]:
```suggestion
// registers a delete gitRepo fuction to ensure the gitRepo is removed cleanly.
```

**Review by anupama2501 — COMMENTED** [review]
tiny nit, lgtm otherwise.

**Review by anupama2501 — APPROVED** [review]

**Review by igomez06 — APPROVED** [review]
LGTM do we need backports?

**Comment by slickwarren** [comment]
yup, was waiting for your approval. Will start on the backports now

**Pull request merged**
